src/libaktualizr/uptane/fetcher.h

```diff
@@ -30,12 +30,12 @@ class Fetcher {
  public:
   Fetcher(const Config& config_in, std::shared_ptr<INvStorage> storage_in, HttpInterface& http_in)
       : http(http_in), storage(std::move(storage_in)), config(config_in) {}
-  bool fetchMeta();
-  bool fetchRoot(bool director, Version version);
-  bool fetchTarget(const Target& target);
+  // bool fetchMeta();
+  // bool fetchRoot(bool director, Version version);
+  bool fetchVerifyTarget(const Target& target);
+  bool fetchRole(std::string* result, RepositoryType repo, Uptane::Role role, Version version = Version());
```

Comment:
API design remark: maybe not use an argument with default value here, it obfuscates a bit the part of the code that uses it (eg: `updateDirectorMeta` in `sotauptaneclient.cc`).

Defining a separate proxy method `fetchRoleLatest` can also help in this direction

Reply:
@lbonn Thanks, I'll add it to my list for refactoring of refactoring (really small one!). Another thing it that invstorage will define two different functions to store root and non-root metadata (because we only need versioning for the former) and Uptane::DirectorRepository and Uptane::ImagesRepository will be local to uptaneIteration method (will make SotaUptaneClient constructor not wo crowded too).

Reply:
No, looks like I can't make them local once we've separated fetching metadata from downloading images. Can make them class members anyway.